tiledb/sm/filter/filter_create.cc

```diff
@@ -120,61 +127,80 @@ tiledb::sm::FilterCreate::deserialize(
       int compression_level;
       st = (buff->read(&compressor_char, sizeof(uint8_t)));
       if (!st.ok()) {
-        return {st, nullopt};
+        throw std::runtime_error(
+            "[FilterCreate::deserialize] Failed to load compressor");
       }
+      try {
+        ensure_compressor_is_valid(compressor_char);
+      } catch (const std::exception& e) {
+        std::throw_with_nested(
+            std::runtime_error("[FilterCreate::deserialize] "));
+      }
+
+      /* Note: compression level is directly handled by the Compressor
+        and thus needn't be validated here. */
       Compressor compressor = static_cast<Compressor>(compressor_char);
       st = buff->read(&compression_level, sizeof(int32_t));
       if (!st.ok()) {
-        return {st, nullopt};
+        throw std::runtime_error(
+            "[FilterCreate::deserialize] Failed to load compression level");
       }
-      return {Status::Ok(),
-              make_shared<CompressionFilter>(
-                  HERE(), compressor, compression_level, version)};
+      return make_shared<CompressionFilter>(
+          HERE(), compressor, compression_level, version);
     }
     case FilterType::FILTER_BIT_WIDTH_REDUCTION: {
       uint32_t max_window_size;
       st = buff->read(&max_window_size, sizeof(uint32_t));
       if (!st.ok()) {
-        return {st, nullopt};
+        throw std::runtime_error(
+            "[FilterCreate::deserialize] Failed to load maximum window size");
       }
-      return {Status::Ok(),
-              make_shared<BitWidthReductionFilter>(HERE(), max_window_size)};
+      if (max_window_size > 1024) {
```

Comment:
Note: there's not currently an enforced "highest max_window_size" - this will be changed to a `// TODO Add validation` for the future.